apps/framework-cli/src/framework/core/infrastructure_map.rs

```diff
@@ -1314,11 +1315,99 @@ impl InfrastructureMap {
                 &project.project_location,
             )
             .await?;
-            todo!("transform the output to infra map, maybe with a intermediate struct")
+            let json = serde_json::to_string(&_objects)?;
+            log::debug!("load_from_user_code inframap json: {}", json);
+
+            Self::from_json_string(&json).map_err(|e| {
+                anyhow::anyhow!("Failed to parse infrastructure map from TypeScript: {}", e)
+            })
         } else {
             todo!("Python implementation")
         }
     }
+
+    pub fn from_json_string(json: &str) -> Result<Self, serde_json::Error> {
+        let partial: PartialInfrastructureMap = serde_json::from_str(json)?;
```

Comment:
you can use `serde_json::from_value` to avoid the work of parsing from string twice